encoding/v2/kzg/prover/backend/gnark/multiframe_proof.go

```diff
@@ -94,8 +97,16 @@ func (p *KzgMultiProofBackend) ComputeMultiFrameProofV2(
 
 	firstECNttDone := time.Now()
 
-	// last step (5) "take first d elements of h^ as h"
-	h := sumVecInv[:dimE]
+	// last step (5) "take first d elements of h^ as h
+	h := sumVecInv[:len(sumVecInv)/2]
+
+	// append identity to prepare the vector which can be taken FFT for erasure coding
+	identity := bn254.G1Affine{}
+	identity.SetInfinity()
+	// now extend h with padding to do erasure coding on the proof
+	for i := uint64(len(h)); i < numTotalChunks; i++ {
+		h = append(h, identity)
+	}
```

Comment:
you could do like in the icicle, and just prepopulate an identity slice with the correct elements, and then you'd only need to append once, which would be more efficient.